src/uberjob/progress/_simple_progress_observer.py

```diff
@@ -73,14 +71,16 @@ def to_progress_string(self):
 
 
 class State:
-    def __init__(self):
-        self.section_scope_mapping = defaultdict(lambda: defaultdict(ScopeState))
+    def __init__(self, start_time):
+        self.section_scope_mapping = {}
         self.running_count = 0
         self._running_section_scopes = set()
-        self._prev_time = None
+        self._prev_time = start_time
 
     def increment_total(self, section, scope, amount: int):
```

Comment:
Increment total is called just a few times, and it makes sense to use setdefault here and avoid use of defaultdict entirely.